distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/InstallPluginAction.java

```diff
@@ -268,7 +275,18 @@ private static void handleInstallXPack(final Build.Flavor flavor) throws UserExc
     private Path download(PluginDescriptor plugin, Path tmpDir) throws Exception {
         final String pluginId = plugin.getId();
 
-        if (OFFICIAL_PLUGINS.contains(pluginId)) {
+        // See `InstallPluginCommand` it has to use a string argument for both the ID and the URL
+        if (OFFICIAL_PLUGINS.contains(pluginId) && (plugin.getUrl() == null || plugin.getUrl().equals(pluginId))) {
+            final String pluginArchiveDir = System.getenv("ELASTICSEARCH_PLUGIN_ARCHIVE_DIR");
+            if (pluginArchiveDir != null && pluginArchiveDir.isEmpty() == false) {
+                final Path pluginPath = getPluginArchivePath(pluginId, pluginArchiveDir);
+                if (Files.exists(pluginPath)) {
+                    terminal.println("-> Downloading " + pluginId + " from local archive: " + pluginArchiveDir);
+                    return downloadZip("file:" + pluginPath, tmpDir);
```

Comment:
Shouldn't this be `"file://" + pluginPath`? I don't think `file:/path/to/file` is a valid URI.

Reply:
I had to loop this up, and according to [Wikipedia](https://en.wikipedia.org/wiki/File_URI_scheme) it is a valid variant, and `URI` parses it fine. But I changed it anyway.

Reply:
Oh, interesting. :+1: